ygen/protogen.go

```diff
@@ -639,9 +649,23 @@ func safeProtoIdentifierName(name string) string {
 // between 1 and 2^29-1. The values 19,000-19,999 are excluded as these are explicitly
 // reserved for protobuf-internal use by https://developers.google.com/protocol-buffers/docs/proto3.
 func protoTagForEntry(e *yang.Entry) (uint32, error) {
-	// TODO(robjs): Replace this function with the final implementation
-	// once concluded.
-	return 1, nil
+	return fieldTag(e.Path())
+}
+
+// fieldTag takes an input string and calculates a FNV hash for the value. If the
+// hash is in the range 19,000-19,999 or 1-1,000, the input string has _ appended to
+// it and the hash is calculated.
+func fieldTag(s string) (uint32, error) {
```

Comment:
People might wonder why FNV was chosen. I guess because it's simple?

Reply:
I'm planning to publish the document showing the behaviour of different hash functions that I considered here. It's sufficiently simple and it provides good enough hashing to ensure that we do not get field number clashes.